Akka.Serialization.MessagePack/Internal/IntIndexedMessagePackFormatterDict.cs

```diff
@@ -56,20 +56,21 @@ public IMessagePackFormatter TryAdd(int i, IMessagePackFormatter formatter)
             //We care about two things in the lock.
             //First, if we need a resize, we do the resize,
             //       and just copy the instance over because we have lock.
+            //Also, Guard for edge case where we get a type that misses bounds,
+            //      i.e. concurrent systems in same process.
             //Otherwise, we check whether we are there
             //           (if a competitor added it on resize, we see via fence)
             //           and if not, add ours.
-            if (_formatters.Length <= i)
+            while (_formatters.Length <= i)
```

Comment:
This is an edge case, so I think it's OK to have the pain of multiple realloc and copies for now, we can create separate issue to resolve if desired. It should only come up if something along the lines of multiple actorsystems start and deserialize in the wrong order.